### PR TITLE
add quack to pybm

### PR DESCRIPTION
**Pull request by liqiangxl**
This PR only added cross entropy fwd, it benchmarks `F.cross_entropy(logits, labels, reduction="none")` same as [quack](https://github.com/Dao-AILab/quack/blob/7dcb92ecd541d64e36cd5689cc700bf4a4c10652/benchmarks/benchmark_cross_entropy.py#L37).

### CODE REVIEW

**Comment by liqiangxl** [comment]
!test

File: benchmarks/python/test_cross_entropy_loss.py [hunk 1]
```diff
@@ -125,3 +127,103 @@ def fwd_call(inp):
     outputs = fwd_fn(inputs)
     grads = SyntheticMiniModel.grads()
     run_benchmark(benchmark, unary_bwd_torch, [outputs, grads, *inputs])
+
+
+# Simple test of F.cross_entropy without final reduction
+# thunder won't use nvFuser for this fusion, here we manually define the fusion
```
Comment (Priya2698) [inline benchmarks/python/test_cross_entropy_loss.py:133]: Does Thunder send this to another executor over nvfuser?
Or do you mean that there is something missing in the Thunder-nvfuser path to support this fusion correctly?

Comment (liqiangxl) [reply benchmarks/python/test_cross_entropy_loss.py:133]: It's by design to not use nvFuser. Can be changed. Here is the current specific requirement to user nvFuser. https://github.com/Lightning-AI/lightning-thunder/blob/35547b63dd42f13080a3c3d6c0c1d66e877a25ad/thunder/executors/nvfuserex_impl.py#L2843C5-L2843C25

Comment (Priya2698) [reply benchmarks/python/test_cross_entropy_loss.py:133]: I am still not clear. Which condition would not be satisfied currently?

> Can be changed.

Yes, we should update that instead if nvfuser now supports a case that was previously missing.

Comment (protonu) [reply benchmarks/python/test_cross_entropy_loss.py:133]: We optimize for special cases in nvFuser otherwise we don't handle cross-entropy loss.

Comment (Priya2698) [reply benchmarks/python/test_cross_entropy_loss.py:133]: So is the benchmark aimed at testing prior to supporting within Thunder?

**Review by protonu — APPROVED** [review]

File: benchmarks/python/test_cross_entropy_loss.py (hunk 1 above)
Comment (protonu) [reply benchmarks/python/test_cross_entropy_loss.py:133]: I think @liqiangxl will be better person to answer that. 
I had added support in Thunder, but that was for certain cases to optimize the computation.

Comment (liqiangxl) [reply benchmarks/python/test_cross_entropy_loss.py:133]: > So is the benchmark aimed at testing prior to supporting within Thunder?

This benchmark is testing `F.cross_entropy(logits, labels, reduction="none")` which is not supported by thunder/nvFuser yet. We want to compare our performance with CuTe DSL, the current reference implementation in CuTe DSL only supports this version.

Comment (liqiangxl) [reply benchmarks/python/test_cross_entropy_loss.py:133]: `F.cross_entropy(logits, labels, reduction="none")` is not a popular usage in actual training. It just happens the reference implementation used this one.

Comment (Priya2698) [reply benchmarks/python/test_cross_entropy_loss.py:133]: Thanks @liqiangxl. Can you put this as a comment? The original comment made me think that we could go via Thunder, but don't currently.

File: benchmarks/python/test_cross_entropy_loss.py [hunk 2]
```diff
@@ -125,3 +127,118 @@ def fwd_call(inp):
...
+    if not disable_benchmarking:
+        run_benchmark(benchmark, fd.execute, inputs)
+
+
+def quack_cross_entropy_fwd_wrapper(inputs: list):
+    return quack_cross_entropy_fwd(*inputs, return_dx=False)
+
+
+@pytest.mark.parametrize(
+    "executor", ["eager", "torchcompile", "thunder", "thunder-torchcompile", "quack"]
```
Comment (Priya2698) [inline benchmarks/python/test_cross_entropy_loss.py]: How is Thunder going to schedule this then?
The executor `Thunder` uses nvfuser.

File: benchmarks/python/test_cross_entropy_loss.py (hunk 1 above)
Comment (liqiangxl) [reply benchmarks/python/test_cross_entropy_loss.py:133]: > I am still not clear. Which condition would not be satisfied currently?
> 
> > Can be changed.
> 
> Yes, we should update that instead if nvfuser now supports a case that was previously missing.

(1) data type 
(2) reduction

File: benchmarks/python/test_cross_entropy_loss.py (hunk 2 above)
Comment (liqiangxl) [reply benchmarks/python/test_cross_entropy_loss.py]: cross entropy in thunder is split into `log_softmax` and `nll_loss` https://github.com/Lightning-AI/lightning-thunder/blob/dbad41c148c4220131efe8097ee0937b3d44bc5a/thunder/torch/__init__.py#L5165

Comment (Priya2698) [reply benchmarks/python/test_cross_entropy_loss.py]: Got it, so in this case, the shapes are not the same? It would also be useful to provide this reference in a comment in the file. As is, it might be misleading to see both Thunder and nvfuser being benchmarked, while we claim that Thunder cannot use nvfuser.

> cross entropy in thunder is split into log_softmax and nll_loss

How does that fusion differ from the manual fusion in the PR? Are they still computing the same thing? (your comment on nvfuser definition mentions `without final reduction`)

Comment (Priya2698) [reply benchmarks/python/test_cross_entropy_loss.py]: @liqiangxl can you clarify this?

Essentially, I want to understand:
1. How are nvfuser and Thunder fusions different?
2. Are they both measuring the same thing?

Comment (liqiangxl) [reply benchmarks/python/test_cross_entropy_loss.py]: 1. How are nvfuser and Thunder fusions different?
Thunder won't generate a fusion for nvFuser. It goes to torch eager using log_softmax and nll_loss

3. Are they both measuring the same thing?
Yes, all are doing the same computation equivalent to `F.cross_entropy(logits, labels, reduction="none")`

Comment (Priya2698) [reply benchmarks/python/test_cross_entropy_loss.py]: We should remove `Thunder` executor (should rename to `Thunder-nvfuser`) then. That is supposed to benchmark `nvFuser` through `Thunder`. We are already benchmarking eager separately.

**Comment by liqiangxl** [comment]
!build
